service/cachemdw/keys.go

```diff
@@ -0,0 +1,67 @@
+package cachemdw
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/kava-labs/kava-proxy-service/decode"
+)
+
+type CacheItemType int
+
+const (
+	CacheItemTypeQuery CacheItemType = iota + 1
+)
+
+func (t CacheItemType) String() string {
+	switch t {
+	case CacheItemTypeQuery:
+		return "query"
+	default:
+		return "unknown"
+	}
+}
+
+func BuildCacheKey(cacheItemType CacheItemType, parts []string) string {
+	fullParts := append(
+		[]string{
+			cacheItemType.String(),
+		},
+		parts...,
+	)
+
+	return strings.Join(fullParts, ":")
+}
+
+// GetQueryKey calculates cache key for request
+func GetQueryKey(
+	chainID string,
+	req *decode.EVMRPCRequestEnvelope,
+) (string, error) {
+	if req == nil {
+		return "", fmt.Errorf("request shouldn't be nil")
+	}
+
+	// TODO(yevhenii): use stable/sorted JSON serializer
```

Comment:
doesn't the std lib's serializer do that? https://stackoverflow.com/a/18668885/5666959

Reply:
looks like it does, I'll remove TODO